core/tests/protractor_utils/CollectionEditorPage.js

```diff
@@ -72,21 +72,19 @@ var CollectionEditorPage = function() {
     await waitFor.elementToBeClickable(
       addExplorationButton, 'Unable to find exploration: ' + query);
 
-    var matched = false;
     var dropdownResultElement = element(
       by.cssContainingText('.dropdown-menu', new RegExp(query)));
-    if (await dropdownResultElement.isPresent()) {
-      await action.click('DropDown Result Element', dropdownResultElement);
-      matched = true;
-    }
-    if (!matched) {
-      // Press Tab to fill in the default result should one appear when
-      // none of the answer matches the given query.
-      await action.sendKeys(
-        'Add Exploration Input', addExplorationInput, protractor.Key.TAB);
-      // If query gets zero result, hitting Tab would not enable the
-      // addExplorationButton.
+    await waitFor.presenceOf(
+      dropdownResultElement, 'Unable to find exploration: ' + query);
+
+    var matchingSearchResult = element(by.cssContainingText(
+      '.uib-typeahead-match', new RegExp(query)));
+    if (await matchingSearchResult.isDisplayed()) {
```

Comment:
Does this appear instantly? If not we should allow time for it to load (e.g. with a waitFor)

Reply:
It does, but I added a waitFor anyway. Thanks!

Reply:
I don't think this will do what you want. Right now, `matchingSearchResult.isDisplayed()` will always be true because the waitFor above will fail the test if `matchingSearchResult` is not present. I see two options:
1. Drop the waitFor since the element appears instantly
2. Pass a failure handler that causes the waitFor to not fail the test if it fails. We have talked about this for a while, but I don't think we've ever worked out how to do it

Reply:
That's odd, though. One of the e2e tests (collections suite) tests exactly this scenario, and it passes. Is that unexpected?

Reply:
(It's the "should visit the collection editor" one, which adds collections both by their ID and their title.)

Reply:
Is the `Testing` search the one you expect to cause `matchingSearchResult` to be missing? I notice that `Testing` is in an exploration description. Does the search only cover titles?

Another possibility: Could `matchingSearchResult` be present but invisible?

Reply:
No -- I'm referring to the one a few lines up, which is adding an exploration by ID.

But ... oh! I think I know what's going on, that doesn't actually use the "search" method at all. So I think you're correct here. I'll work on fixing it. Thanks!

Reply:
Fixed, PTAL. Thanks for catching this.